assets/js/components/ChecksCatalog/CatalogContainer.jsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+
+import { Disclosure, Transition } from '@headlessui/react';
+
+import ReactMarkdown from 'react-markdown';
+import remarkGfm from 'remark-gfm';
+
+import NotificationBox from '@components/NotificationBox';
+import LoadingBox from '@components/LoadingBox';
+
+import { groupBy } from '@lib/lists';
+
+import { EOS_ERROR } from 'eos-icons-react';
+
+const CatalogContainer = ({
+  getCatalog = () => {},
+  catalogData = [],
+  catalogError = null,
+  loading = false,
+}) => {
+  if (loading) {
+    return <LoadingBox text="Loading checks catalog..." />;
+  }
+
+  if (catalogError) {
+    return (
+      <NotificationBox
+        icon={<EOS_ERROR className="m-auto" color="red" size="xl" />}
+        text={catalogError}
+        buttonText="Try again"
+        buttonOnClick={getCatalog}
+      />
+    );
+  }
+
+  return (
+    <div>
+      {Object.entries(groupBy(catalogData.items, 'group')).map(
+        ([group, checks], idx) => (
+          <div
+            key={idx}
+            className="check-group bg-white shadow overflow-hidden sm:rounded-md mb-8"
+          >
+            <div className="bg-white px-4 py-5 border-b border-gray-200 sm:px-6">
+              <h3 className="text-lg leading-6 font-medium text-gray-900">
+                {group}
+              </h3>
+            </div>
+            <ul role="list" className="divide-y divide-gray-200">
+              {checks.map((check) => (
+                <li key={check.id}>
+                  <Disclosure>
+                    <Disclosure.Button
+                      as="div"
+                      className="flex justify-between w-full cursor-pointer hover:bg-gray-100"
+                    >
+                      <div className="check-row px-4 py-4 sm:px-6">
+                        <div className="flex items-center">
+                          <p className="px-2 inline-flex text-xs leading-5 font-semibold rounded-full bg-green-100 text-green-800">
+                            {check.id}
+                          </p>
+                          {check.premium > 0 && (
+                            <p className="px-2 inline-flex text-xs leading-5 font-semibold rounded-full bg-green-100 text-green-800">
+                              Premium
+                            </p>
+                          )}
+                        </div>
+                        <div className="mt-2 sm:flex sm:justify-between">
+                          <div className="sm:flex">
+                            <ReactMarkdown
+                              className="markdown text-sm"
+                              remarkPlugins={[remarkGfm]}
+                            >
+                              {check.description}
+                            </ReactMarkdown>
```

Comment:
Do we wanna extract all this markup in some more granular components?

Reply:
Do you mean to create our own markdown component with those defaults? 
In this case `text-sm` would be an additional style.

Reply:
Can we do the next? Let me merge this as it is about this topic, and I will create other PR with a new markdown component and use it all around the code. What do you think?

Reply:
Not the markdown, I mean the whole HTML body of this component :sweat_smile: 

In a similar way to https://github.com/trento-project/web/blob/main/assets/js/components/ChecksResults/ChecksResults.jsx, we can make this code more readable just having a wrapper component for the `div`s and the `ul` and another component for the `li`

Reply:
Anyway if you wanna do that in a subsequent PR, feel free.

Reply:
@dottorblaster CheckItem extracted to an individual component. Going further looks like an overkill to me